pkg/util/ipset/ipset.go

```diff
@@ -72,74 +72,149 @@ var EntryMemberPattern = "(?m)^(.*\n)*Members:\n"
 // ipset version output is similar to "v6.10".
 var VersionPattern = "v[0-9]+\\.[0-9]+"
 
+// Option is variadic initializer for IPSet
+type Option func(set *IPSet)
+
+// NewIPSet returns a new IPSet which is an Interface to a set
+func NewIPSet(name string, opts ...Option) (*IPSet, error) {
+	ret := IPSet{
+		name: name,
+	}
+	setIPSetDefaults(&ret)
+	for _, opt := range opts {
+		opt(&ret)
+	}
+	if err := ret.validate(); err != nil {
+		return &IPSet{}, err
+	}
+	return &ret, nil
+}
+
 // IPSet implements an Interface to a set.
 type IPSet struct {
-	// Name is the set name.
-	Name string
-	// SetType specifies the ipset type.
-	SetType Type
-	// HashFamily specifies the protocol family of the IP addresses to be stored in the set.
+	// name is the set name.
+	name string
+	// setType specifies the ipset type.
+	setType Type
+	// hashFamily specifies the protocol family of the IP addresses to be stored in the set.
 	// The default is inet, i.e IPv4.  If users want to use IPv6, they should specify inet6.
-	HashFamily string
-	// HashSize specifies the hash table size of ipset.
-	HashSize int
-	// MaxElem specifies the max element number of ipset.
-	MaxElem int
-	// PortRange specifies the port range of bitmap:port type ipset.
-	PortRange string
+	hashFamily string
+	// hashSize specifies the hash table size of ipset.
+	hashSize int
+	// maxElem specifies the max element number of ipset.
+	maxElem int
+	// portRange specifies the port range of bitmap:port type ipset.
+	portRange string
 	// comment message for ipset
-	Comment string
+	comment string
+}
+
+// HashFamily set hashFamily to IPSet
```

Comment:
Each option should document any preconditions - like "This applies to any set type which uses a hash.  If not specified, the default hash family is xyz" or something